### PR TITLE
Added `architect login` cmd

### PR DESCRIPTION
**Pull request by tjhiggins**
Handles the auth0 password grant
Stores access_token in the native credentials store
Created custom base Command
Added ArchitectClient for communication with hub/registry

### CODE REVIEW

File: src/commands/login.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    try {
+      await this.login(answers.username, answers.password);
+    } catch (err) {
+      this.error(_error(err.message));
+    }
+  }
+
+  async promptOptions() {
+    const {flags} = this.parse(Login);
+    return inquirer.prompt([{
```
Comment (davidthor) [inline src/commands/login.ts]: Let's skip the prompt if the flags are passed in during command execution. That would help logins on headless flows like CI pipelines.

**Comment by davidthor** [comment]
Looks very straitforward. We can merge once we store the token as docker creds too

File: src/common/protoc-executor.ts [hunk 2]
```diff
@@ -43,17 +43,26 @@ namespace ProtocExecutor {
...
-      '--user', process.platform === 'win32' ? '1000:1000' : '$(id -u):$(id -g)',  // TODO figure out correct user for windows
-      'architectio/protoc-all',
-      '-f', `${mounted_proto_path}`,
-      '-i', mount_dirname,
-      '-l', target_language,
-      '-o', MANAGED_PATHS.DEPENDENCY_STUBS_DIRECTORY
-    ].join(' '), {stdio: 'ignore'});
+
+    // execSync caused the output logs to hang
+    await new Promise(resolve => {
```
Comment (davidthor) [inline src/common/protoc-executor.ts]: Why this change if we’re just going to await the result anyway?

File: src/commands/services/generate.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { flags } from '@oclif/command';
```
Comment (davidthor) [inline src/commands/services/generate.ts]: Looks like there are two new files here that are unrelated to login. Let’s pull those into a separate PR.

File: src/common/protoc-executor.ts (hunk 2 above)
Comment (tjhiggins) [reply src/common/protoc-executor.ts]: execSync was causing the spinners to hang for whatever reason even if it was in an async func. For whatever reason await exec causes it to behave as expected.

File: src/commands/services/generate.ts (hunk 3 above)
Comment (tjhiggins) [reply src/commands/services/generate.ts]: More than 2. I can pull them out.
